### ISSUE TITLE
Automatic address guessing fails on fedora 20, centos 7

### ISSUE THREAD

**Issue by temeo** [mariadb-corporation/mysql-wsrep#45]
Log

```
2015-01-23 11:33:43 795 [ERROR] WSREP: Process completed with error: /sbin/ifconfig | grep -E '^[[:space:]]+inet addr:' | grep -m1 -v 'inet addr:127' | sed 's/:/ /' | awk '{ print $3 }': 2 (No such file or directory)
2015-01-23 11:33:43 795 [Warning] WSREP: Failed to guess base node address. Set it explicitly via wsrep_node_address.
```

The reason for this is different form of ifconfig output:

```
# ifconfig
eth0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 192.168.3.76  netmask 255.255.255.0  broadcast 192.168.3.255
        inet6 fe80::97:1fff:fefc:1b80  prefixlen 64  scopeid 0x20<link>
        ether 02:97:1f:fc:1b:80  txqueuelen 1000  (Ethernet)
        RX packets 1170  bytes 101571 (99.1 KiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 871  bytes 188128 (183.7 KiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

lo: flags=73<UP,LOOPBACK,RUNNING>  mtu 65536
        inet 127.0.0.1  netmask 255.0.0.0
        inet6 ::1  prefixlen 128  scopeid 0x10<host>
        loop  txqueuelen 0  (Local Loopback)
        RX packets 0  bytes 0 (0.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 0  bytes 0 (0.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
```

Address guessing should be changed to use system calls instead of shell utilities for better portability.

**Comment by philip-galera** [comment]
Should I close https://github.com/codership/galera/issues/123 ?

**Comment by temeo** [comment]
This is mysql-wsrep issue, so discussion should be moved here.

**Comment by nirbhayc** [comment]
@temeo : IP guessing has been reimplemented using system calls in maria.
http://bazaar.launchpad.net/~maria-captains/maria/maria-10.0-galera/revision/3917
http://bazaar.launchpad.net/~maria-captains/maria/maria-10.0-galera/revision/3922

**Comment by alexander-krotov** [comment]
Following condition:

``` c++
    421     if (strcmp(buf, "127.0.0.1") == 0) // lame
    422        continue;
    423      }
```

Could be changed to:

``` c++
   394        // Skip loopback interfaces (like lo:127.0.0.1)
   395        if (ifa->ifa_flags & IFF_LOOPBACK) {
   396           continue;
   397        }
```

**Comment by temeo** [comment]
Fix pushed to 5.5 and 5.6.

Thanks @nirbhayc, @alexander-krotov
